mobius/examples/balanced_brackets.rs

```diff
@@ -0,0 +1,115 @@
+use std::{error::Error, f64::consts::PI};
+
+use mobius::{
+    cline_arc::ClineArc,
+    integer_arcs::{integer_arc_by_direction, integer_arc_by_hemisphere, Hemisphere},
+    rendering::Style,
+    rotation,
+    svg_plot::{render_views, style_geometry, View},
+    transformable::{ClineArcTile, Transformable},
+    translation, Complex,
+};
+
+#[derive(Debug, thiserror::Error)]
+enum BracketError {
+    #[error("Characters must be [ or ], got {0}")]
+    InvalidCharacter(char),
+    #[error("Unbalanced brackets: {0}")]
+    Unbalanced(String),
+}
+
+#[derive(PartialEq, Clone, Copy)]
+enum Bracket {
+    Left,
+    Right,
+}
+
+struct BalancedBrackets {
+    brackets: Vec<Bracket>,
+}
+
+impl BalancedBrackets {
+    pub fn new(bracket_string: &str) -> Result<Self, BracketError> {
+        let mut depth: i32 = 0;
+        let mut brackets = Vec::new();
+        for c in bracket_string.chars() {
+            match c {
+                '[' => {
+                    brackets.push(Bracket::Left);
+                    depth += 1;
+                }
+                ']' => {
+                    brackets.push(Bracket::Right);
+                    depth -= 1;
+                }
+                x => return Err(BracketError::InvalidCharacter(x)),
+            }
+            if depth < 0 {
+                return Err(BracketError::Unbalanced(String::from(bracket_string)));
+            }
+        }
+
+        if depth != 0 {
+            return Err(BracketError::Unbalanced(String::from(bracket_string)));
+        }
+
+        Ok(Self { brackets })
+    }
+
+    pub fn len(&self) -> usize {
+        self.brackets.len()
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = &Bracket> {
+        self.brackets.iter()
+    }
+}
+
+pub fn main() -> Result<(), Box<dyn Error>> {
+    let north_brackets = BalancedBrackets::new("[][[]][][][[[]][]]")?;
```

Comment:
Let me try using `clap` to parse the pattern type `line|circle` and 2 sets of balanced brackets. Maybe also a `--suffix` flag so the output would be `balanced_brackets_<line|circle>_<suffix>.svg`

Then either generate a pattern of integer arcs like below, or ones on the unit circle.

The SVG rendering code will look a little different from one to the next to get the view right.